payjoin/src/psbt/merge.rs

```diff
@@ -0,0 +1,168 @@
+//! Utilities for merging unique v0 PSBTs
+use bitcoin::Psbt;
+
+#[allow(dead_code)]
+/// Try to merge two PSBTs
+/// PSBTs here should not have the same unsigned tx
+/// if you do have the same unsigned tx, use `combine` instead
+/// Note: this method does not merge non inputs or outputs
+/// Note: if there are duplicate inputs, the first input will be kept
+/// Note: if there are duplicate outputs, both outputs will be kept
+/// ```no_run
+/// let psbts = vec![psbt_1.clone(), psbt_2.clone(), ..., psbt_n.clone()];
+/// let merged_psbt = psbts.into_iter().reduce(merge_unsigned_tx).unwrap();
+/// ```
+pub(crate) fn merge_unsigned_tx(acc: Psbt, psbt: Psbt) -> Psbt {
+    let mut unsigned_tx = acc.unsigned_tx;
+    unsigned_tx.input.extend(psbt.unsigned_tx.input);
+    unsigned_tx.input.dedup_by_key(|input| input.previous_output);
+    unsigned_tx.output.extend(psbt.unsigned_tx.output);
+
+    Psbt::from_unsigned_tx(unsigned_tx).expect("pulling from unsigned tx above")
```

Comment:
As soon as this gets used it needs to be handled. Makes little sense to panic. Can just return Result<Psbt, bitcoin::psbt::Error>

Reply:
relevant comment: https://github.com/payjoin/rust-payjoin/pull/523#discussion_r1942222823